core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -580,6 +563,23 @@ object ConfigCommand extends Logging {
     }
   }
 
+  private def alterResourceConfig(adminClient: Admin, entityTypeHead: String, entityNameHead: String, configsToBeDeleted: Seq[String], configsToBeAdded: Map[String, ConfigEntry], resourceType: ConfigResource.Type): Unit = {
+    val oldConfig = getResourceConfig(adminClient, entityTypeHead, entityNameHead, includeSynonyms = false, describeAll = false)
```

Comment:
In reviewing it, I notice that `admin.describeConfigs` will return static configs even if the group is nonexistent. IMHO, that should return GROUP_ID_NOT_FOUND. @AndrewJSchofield @DL1231 WDYT? I can file a jira to fix it if it is NOT expected behavior.

Reply:
That's probably the best behaviour.

Reply:
> This is not wrong because the group could be create any time and it is, it would have those configurations by default.

My concern was the consistency. The type `TOPIC` has similar attributes but we return exception if it is nonexistent [0]. However, maybe we can add enough docs to the APIs to remind users




[0] https://github.com/apache/kafka/blob/c5e91546721b6b160732667ca7450d22927cc3b3/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala#L1513

Reply:
https://issues.apache.org/jira/browse/KAFKA-17359 is opened